### PR TITLE
SA-1732: Fix Summary Report precision & date range with metrics rollup on initial load

### PR DESCRIPTION
**Pull request by kchu1337**
### What Changed
 - Fixed precision being set to 1 min upon initial load in OG metrics rollup summary page.
 - Deleted unnecessary datepicker file.
 - Precision selector now shows correct selected precision. 

### How To Test
 - enable metrics-rollup and go to OG summary report page. Verify that it works.
 - Go to hibana report builder and open the date picker. Pick different date ranges and verify that the precision selector changes based on restricted/recommended precisions.
 - Go to Blocklist incidents collection and verify the date picker there still works.

### CODE REVIEW

File: src/helpers/metrics.js [hunk 1]
```diff
@@ -123,10 +123,14 @@ export function getRecommendedRollupPrecision(from, to = moment()) {
  */
 export function getPrecisionOptions(from, to = moment()) {
   const diff = to.diff(from, 'minutes');
-  return rollupPrecisionMap
+  return _getPrecisionOptions(diff);
+}
+const _getPrecisionOptions = _.memoize(diff => {
```
Comment (kchu1337) [inline src/helpers/metrics.js:128]: Found out, the UI was making this call multiple times, so I memoized it.

File: src/pages/reportBuilder/components/DateTimeSection/PrecisionSelector.js [hunk 2]
```diff
@@ -8,9 +8,11 @@ const PrecisionSelector = ({ from, to, changeTime, selectedPrecision, disabled,
   const precisionOptions = getPrecisionOptions(moment(from), moment(to));
 
   useEffect(() => {
-    const updatedPrecision = getRollupPrecision({ from, to, precision: selectedPrecision });
-    if (updatedPrecision !== selectedPrecision && ready) {
-      changeTime({ precision: updatedPrecision });
+    if (from && to && selectedPrecision) {
```
Comment (kchu1337) [inline src/pages/reportBuilder/components/DateTimeSection/PrecisionSelector.js:11]: Added this, so the side-effect doesn't get called before the reportOptions gets populated by URL params or default params.

File: src/components/datePicker/DatePickerNew.js [hunk 3]
```diff
@@ -93,6 +98,9 @@ const datePickerReducer = (state, { type, payload }) => {
 };
 export function DatePicker(props) {
   const [state, dispatch] = useReducer(datePickerReducer, initialState);
+  const { isSelecting, selected, isDatePickerOpen } = state;
+
+  const getPrecision = props.useMetricsRollup ? getRollupPrecision : getRawPrecision;
```
Comment (kchu1337) [inline src/components/datePicker/DatePickerNew.js:103]: Precision stuff was added to the other datepicker in a previous ticket and I am adding them to this one now.

File: src/components/datePicker/DatePickerNew.js [hunk 4]
```diff
@@ -123,19 +131,30 @@ export function DatePicker(props) {
...
       cancelDatePicker();
     }
 
-    if (!state.selecting && e.key === 'Enter') {
+    if (!isSelecting && e.key === 'Enter') {
       handleSubmit();
     }
   };
 
+  const { updateShownPrecision } = props;
```
Comment (kchu1337) [inline src/components/datePicker/DatePickerNew.js:147]: This will update the precision selector to show the correct precision while selecting a new date.

**Review by nicklemmon — COMMENTED** [review]
@kchu1337:

- [x] The OG summary report bug is fixed
- [x] The datepicker still works on the Blocklist feature

I'm confused about how the precision `<select/>` updates when using the datepicker? When I click between the options on the lefthand side of the datepicker the precision value changes, but seemingly randomly. I don't quite understand what the requirement is.

File: src/components/datePicker/DatePickerNew.js [hunk 5]
```diff
@@ -37,6 +41,7 @@ const ActionListWrapper = styled.div`
 
 const DateSelectorWrapper = styled.div`
   width: max-content;
+  max-width: 38.5rem;
```
[the remark this answers is not in the document]
Comment (kchu1337) [reply src/components/datePicker/DatePickerNew.js:44]: Not sure, copied from the other one.

> @kchu1337:
> 
> * [x]  The OG summary report bug is fixed
> * [x]  The datepicker still works on the Blocklist feature
> 
> I'm confused about how the precision `<select/>` updates when using the datepicker? When I click between the options on the lefthand side of the datepicker the precision value changes, but seemingly randomly. I don't quite understand what the requirement is.

If the current selected precision is allowed, keep that precision. If that precision is not allowed, use the recommended precision.

**Review by nicklemmon — CHANGES_REQUESTED** [review]
@kchu1337 I think I would be more comfortable moving forward if we had some automated testing to support this change? That would also help document what the intended requirement is for the feature.

Also, did you see my comment about `max-width: 38.5rem`? Is there a `sizing` design token value we could use there instead?

**Review by nicklemmon — CHANGES_REQUESTED** [review]
@kchu1337 Left a few comments for you - I'm still able to get unusual precision results by clicking between the different date ranges on the lefthand side of the date picker. Not sure if that's expected, or if I am doing something wrong?

File: cypress/tests/integration/signals-analytics/dateAndTimeSection.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        .subtract(1, 'day');
+      cy.get('[data-id="fromDate"]').should('have.value', dayAgo.local().format('YYYY-MM-DD'));
+      cy.get('[data-id="fromTime"]').should('have.value', dayAgo.local().format('h:mma'));
+      //Re-stub so I can get the url params from the second call
+      cy.stubRequest({
+        url: '/api/v1/metrics/deliverability/time-series**/**',
+        fixture: 'metrics/deliverability/time-series/200.get.json',
+        requestAlias: 'getTimeSeries2',
+      });
+      cy.findByText('Apply').click({ force: true });
```
[the remark this answers is not in the document]
Comment (kchu1337) [reply cypress/tests/integration/signals-analytics/dateAndTimeSection.js:90]: Looking at the visual from the cypress runner, I don't think the datepicker is even opening up correctly. So it's in the DOM but out of view, which is why I used `force`

[the remark this answers is not in the document]
Comment (nicklemmon) [reply cypress/tests/integration/signals-analytics/dateAndTimeSection.js:90]: Weird!

File: cypress/tests/integration/signals-analytics/dateAndTimeSection.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+      cy.findByLabelText('Precision').should('have.value', 'day');
+    });
+
+    it('changing date picker values changes the precision correctly (Hibana)', () => {
+      cy.visit(PAGE_URL_HIBANA);
+      cy.findByLabelText('Date Range').click();
+
+      //Check mouseover
+      cy.get('[aria-label="Wed Jan 29 2020"]').click({ force: true });
+      cy.get('[aria-label="Tue Dec 03 2019"]').trigger('mouseover', { force: true });
```
[the remark this answers is not in the document]
Comment (kchu1337) [reply cypress/tests/integration/signals-analytics/dateAndTimeSection.js:69]: Nope, it doesn't seem to work.

File: cypress/tests/integration/signals-analytics/dateAndTimeSection.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (kchu1337) [reply cypress/tests/integration/signals-analytics/dateAndTimeSection.js:90]: Also `cy.findByRole('button', { name: 'Apply' }).click({ force: true });` doesn't seem to work either.

**Review by nicklemmon — APPROVED** [review]
@kchu1337 Let's do it

**Pull request merged**